importer/TnFr.py

```diff
@@ -0,0 +1,138 @@
+import dateparser
+from Monument import Monument, Dataset
+import importer_utils as utils
+import importer as importer
+
+
+class TnFr(Monument):
+
+    def set_street_address(self):
+        """
+        Set the street address.
+
+        Works if the source data contains a digit.
+        Since the place is not included, transpose
+        it from the place name.
+        """
+        if self.has_non_empty_attribute("adresse"):
+            addr_raw = utils.remove_markup(self.adresse)
+            if utils.contains_digit(addr_raw):
+                settlement = utils.remove_markup(self.site)
+                address = "{}, {}".format(addr_raw, settlement)
+                self.add_statement("located_street", address)
+            else:
+                self.add_to_report("adresse", self.adresse, "located_street")
+
+    def set_location(self):
+        """
+        Set the location.
+
+        If present, use the wikilink, otherwise
+        match value against list of settlements.
+        """
+        loc_q = None
+        loc_raw = self.site
+        if utils.count_wikilinks(loc_raw) == 1:
+            loc_q = utils.q_from_first_wikilink("fr", loc_raw)
+        else:
+            loc_dic = self.data_files["settlements"]
+            loc_match = utils.get_item_from_dict_by_key(dict_name=loc_dic,
+                                                        search_term=loc_raw,
+                                                        search_in="itemLabel")
+            if len(loc_match) == 1:
+                loc_q = loc_match[0]
+
+        if loc_q:
+            self.add_statement("location", loc_q)
+        else:
+            self.add_to_report("site", self.site, "location")
+
+    def set_admin_location(self):
+        """
+        Set the admin location.
+
+        Use the iso code of the gouvernorat for exact
+        matching against list.
+        """
+        adm_q = None
+        iso = self.gouvernorat_iso
+        admin_dic = self.data_files["admin"]
+        admin_match = utils.get_item_from_dict_by_key(dict_name=admin_dic,
+                                                      search_term=iso,
+                                                      search_in="iso")
+        if len(admin_match) == 1:
+            adm_q = admin_match[0]
+            self.add_statement("located_adm", adm_q)
+        else:
+            self.add_to_report("gouvernorat_iso",
+                               self.gouvernorat_iso, "located_adm")
+
+    def set_heritage(self):
+        """
+        Set heritage status with optional qualifiers.
+
+        Use 'decret' as 'described at url'.
+        Extract date from 'date' as 'start time'.
+        """
+        quals = {}
+        heritage_item = self.mapping["heritage"]["item"]
+
+        urls = utils.get_external_links(self.decret)
+        if len(urls) == 1:
+            quals["described_at_url"] = urls[0].strip_code()
+
+        date_templates = utils.wparser.parse(self.date).filter_templates()
```

Comment:
is it always the same template?`wondering if we might want to check that the template is the expected one.

Reply:
added check for correct template